File: leetcode/python/000167.two-sum-ii-input-array-is-sorted.py
class Solution:
	# Time complexity: O(n)
	# Space complexity: O(1)
	def twoSum(self, numbers: list[int], target: int) -> list[int]:
		res = []
		l, r = 0, len(numbers) - 1
		while l < r:
			sum = numbers[l] + numbers[r]
			if sum < target:
				l += 1
			elif sum > target:
				r -= 1
			else:
				res = [l+1, r+1]
				break
		return res

File: leetcode/typescript/000167.two-sum-ii-input-array-is-sorted.ts
function twoSum(numbers: number[], target: number): number[] {
  let l = 0;
  let r = numbers.length - 1;

  while (l < r) {
    const sum = numbers[l] + numbers[r];

    if (sum > target) {
      r--;
    } else if (sum < target) {
      l++;
    } else {
      return [l + 1, r + 1];
    }
  }

  return [];
}
// Time complexity: O(n)
// Space complexity: O(1)
